source/glimmer-components/passing-data-to-components.md

```diff
@@ -0,0 +1,231 @@
+Imagine if functions aren't allowed to take arguments: they won't be very
+useful at all, right? Likewise, if we cannot pass data to our Components, they
+would just be fancy partials with limited uses. Fortunately, Ember made it very
+easy to pass data into our Components.
+
+## Passing Attributes
+
+When invoking Components, you can pass along any number of named attributes,
+just like how you can set attributes on regular HTML elements:
+
+```app/templates/contact-us.hbs
+<h5><x-icon type="speech-bubble" size="large" /> Contact Us</h5>
+
+<x-icon type="phone" /> 1-800-EMBER-JS <br>
+<x-icon type="twitter" /> @emberjs <br>
+<x-icon type="map" /> 123 Some Street, Portland OR
+```
+
+These attributes can then be accessed from the Component's template via the
+`attrs` hash:
+
+```app/templates/components/x-icon.hbs
+<x-icon class="icon icon-{{attrs.type}} {{attrs.size}}" />
+```
+
+You might be wondering: what's the point of going through the `attrs` hash?
+Couldn't Ember just expose the attributes as `{{type}}` instead of the nosier
+`{{attrs.type}}`? Hold your thoughts! We will come back to that soon.
+
+In addition to string attributes, sometimes you might need to pass other kind
+of data such as numbers and booleans. You can do this using the curlies syntax
+instead of quotation marks:
+
+```app/templates/step-two.hbs
+<progress-bar value={{2}} max={{10}} animated={{true}} /> Step 2 / 10
```

Comment:
Note that the DOM spec mirrors almost all properties to attributes, so leaving off quotation marks will mostly do what you expect but allow you to pass values (booleans, dates, etc.) instead of strings. This is especially important for things like `value={{name}}`, whose attribute assigns a "default value", but the property updates the `.value`.

Because of the fact that attributes and properties are melded together in the user's mental model already, the DOM has done work to make attribute strings and property values mostly mirror. What this means is if we hew closely to the same fuzzy mental model that the DOM already worked hard to support, things will work with the same caveats as DOM.